rpmio/rpmio.c

```diff
@@ -523,6 +523,28 @@ fprintf(stderr, "*** ufdOpen(%s,0x%x,0%o)\n", url, (unsigned)flags, (unsigned)mo
     return fd;
 }
 
+/* Return number of threads ought to be used for compression based
+   on a parsed value threads (e.g. from w7T0.xzdio or w7T16.xzdio).
+   Value -1 means automatic detection. */
+
+static int
+get_compression_threads (int threads)
+{
+    if (threads == -1)
+	threads = rpmExpandNumeric("%{getncpus}");
+
+#if __WORDSIZE == 32
+    /* Limit thread up to 4 for 32-bit systems.  */
+    if (threads > 4)
+    {
+	threads = 4;
+	rpmlog(RPMLOG_WARNING, "threading compression limited to 4 threads on 32-bit systems\n");
```

Comment:
I think this should be RPMLOG_DEBUG instead to avoid spewage on systems where it actually gets activated.